### DISCUSSION TITLE
Semi / Auto-Advanced Setup?

### DISCUSSION THREAD

**Discussion by oOStroudyOo** [tteck/Proxmox#3086] [Ideas]
Hi,

When running a script I always use advanced settings, but the only thing I actually change per LXC is the IP address, everything else is the same setup for each LXC.

I'm wondering if it's possible to set my own custom defaults and have the scripts use them?
And perhaps the script only prompts when it can't find a custom default for that field.

For example:
- The IPv4 address is always 192.168.1.Container ID/24
- The apt-cacher and gateway is always the same IP address
- The storage location is always the same drive.
- Debian version = latest

Seems like it could be automated.
Maybe I'm getting lazy 😆

**Comment by di5cord20** [comment]
I have a similar setup and would welcome this solution if one becomes available.

**Comment by Mellowlynx** [comment]
I was thinking about this too, I have multiple storage options, so even with default options, it will ask me to confirm on what storage the LXC should be installed on.
It would indeed be nice to have an environment file, where some default can be set.

**Comment by Korenchkin** [comment]
something like init.d source /some/config? set some defaults there and just add pulling that config...and error checking...oh well,not that easy..

**Reply by oOStroudyOo** [reply]
Always seems easy in theory.
Ahh just grab this entry and job done right? 😆 

400 lines of code, half a head of hair lost, and 3 blistered fingers later... 🤣

**Reply by Mellowlynx** [reply]
There is a build file that is used in all script, so might be that it can be added there without other changes to all the scripts, but I'm afraid they also need an update. 🙏

**Comment by alexanderjerome** [comment]
I just made a comprehensive post on the subject with similar suggestions. Let me know what you think! #3280

**Comment by Mellowlynx** [comment]
@tteck Like to know your thoughts.
I now have a separate Dev/testing Proxmox and would love if I could have a helper-script.env file that sets the needed VLAN tag. :pray:

**Reply by tteck** [reply]
My thoughts are... not interested.
